packages/app-utils/src/components/useLoadTablePlugin.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react';
+import {
+  type TablePluginComponent,
+  isTablePlugin,
+  isLegacyTablePlugin,
+} from '@deephaven/plugin';
+import Log from '@deephaven/log';
+import usePlugins from './usePlugins';
+
+const log = Log.module('@deephaven/app-utils/useTablePlugin');
+
+/**
+ * Creates a table plugin loader function.
+ * @returns A function to load a Table plugin element by name
+ */
+export function useLoadTablePlugin(): (name: string) => TablePluginComponent {
+  const plugins = usePlugins();
+
+  const plugin = useCallback(
+    (name: string) => {
+      // First check if we have any plugin modules loaded that match the TablePlugin.
+      const pluginModule = plugins.get(name);
+      if (pluginModule != null) {
+        if (isTablePlugin(pluginModule)) {
+          return pluginModule.component;
+        }
+        if (isLegacyTablePlugin(pluginModule)) {
+          return pluginModule.TablePlugin;
+        }
+      }
+
+      const errorMessage = `Unable to find table plugin ${name}.`;
+      log.error(errorMessage);
+      throw new Error(errorMessage);
+    },
+    [plugins]
+  );
+
+  return plugin;
+}
```

Comment:
The hook just creates the `loadPlugin` function for `IrisGridPanel` props. Since `IrisGridPanel` is a class component, I felt it was too involved of a change for this refactor